src/service/viewer-impl.js

```diff
@@ -835,6 +835,34 @@ export class Viewer {
   }
 
   /**
+   *
+   * @return {!Promise}
+   */
+  replaceShareTrackingFragment(incomingFragment, outgoingFragment) {
+    if (!this.isEmbedded_) {
+      if (this.win.history.replaceState) {
+        let newUrl = this.win.location.href;
+        if (incomingFragment == "") {
+          if (newUrl.indexOf('#') == -1) {
+            newUrl = newUrl + '#.' + outgoingFragment;
+          } else {
+            newUrl = newUrl.replace('#', '#.' + outgoingFragment + '&');
+          }
+        } else {
+          newUrl = newUrl.replace(incomingFragment, outgoingFragment);
+        }
+        this.win.history.replaceState({}, '', newUrl);
```

Comment:
What do you mean by `this.win.location.replace(newUrl);`? Assign a new value to the `this.win.location.href`? Like `this.win.location.href = this.win.location.href.replace(incomingFragment, outgoingFragment)` ?
I want the history to be modified(replaced) so when the user hit back button, he/she will go to the previous page, not staying on the same page(with different fragment) again.